#27 feature/payment feign client

gkwjdtn975 opened the pull request:
<!-- .github/pull_request_template.md -->

## 📝 PR 개요

- RestTemplate 기반 토스 결제 연동을 FeignClient(`TossPaymentClient`) 기반으로 전환했습니다.  
- `PaymentServiceImpl`에서 `RestTemplate` 호출을 모두 Feign client 호출로 대체하였으며,  
  - 결제 생성(`createPayment`),  
  - 결제 확정(`confirmPayment`),  
  - 결제 취소(`cancelPayment`) 로직을 각각 FeignClient 메서드로 호출  
- `ResponseEntity.getBody()`를 사용해 Feign 응답 본문을 추출하도록 수정  
- 기본 인증(Basic Auth) 헤더 생성을 위한 `basicAuthHeader()` 유틸 메서드 추가  
- 관련 테스트 및 DTO, 설정 클래스를 함께 업데이트

## 🔗 관련 이슈

- 없음  

## ✅ 체크리스트

- [x] FeignClient 호출로 결제 API 연동이 정상 동작합니다.  
- [x] 기존 `RestTemplate` 호출 관련 코드를 모두 제거했습니다.  
- [x] 새로운/수정된 기능에 대한 단위 테스트를 추가 또는 수정했습니다.  
- [x] API 연동 설정(`TossPaymentConfig`, `application.yml`)을 검토 및 반영했습니다.  
- [x] 코드 스타일 가이드 및 커밋 메시지 규칙을 준수했습니다.

## 🖼️ 변경 전/후 스크린샷 (UI 변경 시, 선택)

- UI 변경 없음  

## 💬 기타 참고 사항

- `TossPaymentClient` 인터페이스에서 Feign 호출 시 반환 타입이 `ResponseEntity<T>`이므로 `.getBody()` 호출을 꼭 확인해주세요.  
- 인수 테스트 환경에서 Feign Stub 서버를 구성해두었습니다.  
- 머지 후 `application.yml`의 `toss.base-url`이 올바르게 설정되어 있는지 확인해야 합니다.

gkwjdtn975 committed:
fix(payment): 테스트: 결제 모듈 단위 테스트 추가

              - PaymentServiceImplTest: requestTossPayment 정상 흐름 검증 (주문 조회 → Toss API 호출 → DTO 반환)
              - PaymentControllerTest: POST /api/v1/payments/toss 요청 시 201 응답 및 JSON 페이로드 검증

gkwjdtn975 committed:
fix(payment): 테스트: 결제 모듈 단위 테스트 추가

              - PaymentServiceImplTest: requestTossPayment 정상 흐름 검증 (주문 조회 → Toss API 호출 → DTO 반환)
              - PaymentControllerTest: POST /api/v1/payments/toss 요청 시 201 응답 및 JSON 페이로드 검증

gkwjdtn975 committed:
fix(payment): 테스트: 결제 모듈 단위 테스트 추가

              - PaymentServiceImplTest: requestTossPayment 정상 흐름 검증 (주문 조회 → Toss API 호출 → DTO 반환)
              - PaymentControllerTest: POST /api/v1/payments/toss 요청 시 201 응답 및 JSON 페이로드 검증

gkwjdtn975 committed:
fix(payment): 테스트: 결제 모듈 단위 테스트 추가

              - PaymentServiceImplTest: requestTossPayment 정상 흐름 검증 (주문 조회 → Toss API 호출 → DTO 반환)
              - PaymentControllerTest: POST /api/v1/payments/toss 요청 시 201 응답 및 JSON 페이로드 검증

gkwjdtn975 committed:
feat(payment): FeignClient로 Toss 결제 연동 전환

- RestTemplate → FeignClient (`TossPaymentClient`) 사용
- `ResponseEntity.getBody()` 호출로 응답 본문 추출
- `PaymentServiceImpl`에 기본 인증 헤더 생성 로직 추가
- 결제 생성/확정/취소 로직 모두 Feign 호출로 대체

gkwjdtn975 commented:
.

gkwjdtn975 committed:
feat(payment): FeignClient로 Toss 결제 연동 전환

- RestTemplate → FeignClient (`TossPaymentClient`) 사용
- `ResponseEntity.getBody()` 호출로 응답 본문 추출
- `PaymentServiceImpl`에 기본 인증 헤더 생성 로직 추가
- 결제 생성/확정/취소 로직 모두 Feign 호출로 대체

mni-js commented on pom.xml:
```diff
@@ -58,10 +51,12 @@
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
 
+        <!-- Jackson -->
```
spring 기본 설정에 이미 존재하기 때문에 머지 후 없애도 될 듯 합니다.

mni-js commented on pom.xml:
```diff
@@ -83,24 +77,19 @@
         </dependency>
 
-        <!-- Spring Security Web -->
+        <!-- Security -->
```
스프링 시큐리티를 사용하는 곳이 없다면 머지 후 없애도 될 것 같습니다.

mni-js reviewed (approved):
고생하셨습니다. 합치죠

gkwjdtn975 committed:
Merge remote-tracking branch 'origin/develop' into #27-feature/payment-FeignClient

# Conflicts:
#	pom.xml
#	src/main/java/com/nhnacademy/bookstoreorderapi/BookstoreOrderApiApplication.java
#	src/main/java/com/nhnacademy/bookstoreorderapi/payment/service/impl/PaymentServiceImpl.java
#	src/main/resources/application-dev.yml
#	src/main/resources/application.yml

gkwjdtn975 committed:
feat(payment): FeignClient로 Toss 결제 연동 전환

- RestTemplate → FeignClient (`TossPaymentClient`) 사용
- `ResponseEntity.getBody()` 호출로 응답 본문 추출
- `PaymentServiceImpl`에 기본 인증 헤더 생성 로직 추가
- 결제 생성/확정/취소 로직 모두 Feign 호출로 대체
